File: test-suite-kotlin/src/test/kotlin/io/micronaut/kafka/docs/consumer/config/ConsumerGroupIdListener.kt
package io.micronaut.kafka.docs.consumer.config

import io.micronaut.configuration.kafka.annotation.KafkaListener
import io.micronaut.context.annotation.Requires

@Requires(property = "spec.name", value = "ConfigProductListenerTest") // tag::clazz[]
// tag::annotation[]
@KafkaListener(groupId = "myGroup")
// end::annotation[]
class ConsumerGroupIdListener {
    // define topic listeners
}

File: test-suite-python/src/test/python/micronaut/kafka/docs/consumer/config/ConsumerGroupIdListener.py
from micronaut.configuration.kafka.annotation import KafkaListener
from micronaut.context.annotation import Requires


@Requires(property="spec.name", value="ConfigProductListenerTest")
# tag::annotation[]
@KafkaListener(groupId="myGroup")
# end::annotation[]
class ConsumerGroupIdListener:
    # define topic listeners
    pass
